### ISSUE TITLE
Update mPDF and move it to Composer

### ISSUE THREAD

**Issue by pgorod** [salesagility/SuiteCRM#7111]
#### Issue
We use a PDF package called mPDF which I believe comes from here:
https://github.com/mpdf/mpdf

In our codebase, awkwardly, this lives in directory 
`modules/AOS_PDF_Templates/PDF_Lib/`

We seem to be using mPDF version 5.7.1, dated Sep/2013, so over 5 years old. This version is no longer supported. The latest in that 5.7 branch is 5.7.4, from 2014.

The most current version is 8.0.0, it just came out.

#### Problems with PHP 7.x

The version we use doesn't support PHP 7.x, and as more people move to 7.x (which we say we support), the problems start to get more frequent.

One particularly nasty manifestation of this problem is that PHP warnings are getting dumped **inside the generated PDF file**, thus making it corrupt and unopenable. This should probably be fixed regardless of any version updates.

**Comment by lazka** [comment]
I've created #7118 for this

**Comment by pgorod** [comment]
Now that we have a proposed fix (yay 🎉 !) does anybody have a clue about why PHP warnings are dumped inside the generated PDF file? Is this something we need to fix in our code or is it an mPDF problem?

**Comment by lazka** [comment]
No idea. Do you know a way to reproduce the error?

**Comment by pgorod** [comment]
Sort of. I couldn't see this happening in my own installation, so maybe there is some other factor that I am unaware of.

But the basic idea is to make PHP throw a warning like the ones you see here (read through a few posts)

https://suitecrm.com/suitecrm/forum/feedback/22105-print-as-pdf-error

Let's just hope you can get the error to appear for you...

**Comment by lazka** [comment]
Oh, that's because they have `display_errors` on, which injects the error messages as HTML into the php output, and should never be enabled in production.

**Comment by pgorod** [comment]
I agree it should be disabled, but does PHP output have to go into the PDF, is that the only way that mPDF works? I think we should fail more gracefully here. 

Maybe turn off `display_errors` from code before starting the PDF generation, and then setting it back the way t was in the end, I don't know...

**Comment by pgorod** [comment]
@Dillon-Brown let's not neglect the "warnings inside the PDF" Issue, it keeps coming up in the forums lately.

**Comment by Dillon-Brown** [comment]
@pgorod #7322 and #7323 should hopefully resolve the warnings inside the PDF issue.

**Comment by connorshea** [comment]
This is fixed in `develop` now I think?

**Comment by pgorod** [comment]
@Dillon-Brown can you please help me understand the status on this one?

- the current PR isn't merged yet
- PR #7322 is closed without merging, probably because it was aimed at `hotfix-7.8.x` which is EOL
- PR #7323 (`hotfix-7.10.x`) is not merged yet, probably for no reason except lack of time while we were doing other things

So I am seeing another user complaining about the warnings inside PDF files in the Forums, with 7.11.8. 

My question is if #7323 is dependent on this one here for some reason, or if we can go ahead and merge it.

**Comment by Dillon-Brown** [comment]
@pgorod So it looks like the original "Update mPDF and move it to composer" was completed here: #7118 but it can probably be backported to 7.10.x/7.11.x now. 

For the second part of this issue "PHP warnings are getting dumped inside the generated PDF file". It's still a WIP here #7323, not sure where I actually got with it but it should be able to go into hotfix-7.10.x without the composerify PR being included.

**Comment by christf** [comment]
what is the exit criteria for this issue? Yes there was an upgrade from mpdf 5 done to 6. That is great.
8 is current and we need at least 7 to support #9092 So how should we proceed? Have another issue asking for mpdf 7?

**Comment by tsmgeek** [comment]
There re lots of warnings being flagged by `/PDF_Lib/classes/cssmgr.php`, not sure if they may be fixed in later version.

```
[Thu Sep 09 15:35:03.670282 2021] [php7:warn] [pid 996] [client 10.100.106.71:44960] PHP Warning:  Cannot assign an empty string to a string offset in /var/www/html/modules/AOS_PDF_Templates/PDF_Lib/classes/cssmgr.php on line 1248
[Thu Sep 09 15:35:03.670297 2021] [php7:warn] [pid 996] [client 10.100.106.71:44960] PHP Warning:  Illegal string offset 'ID' in /var/www/html/modules/AOS_PDF_Templates/PDF_Lib/classes/cssmgr.php on line 1322
[Thu Sep 09 15:35:03.670302 2021] [php7:warn] [pid 996] [client 10.100.106.71:44960] PHP Warning:  Illegal string offset 'ID' in /var/www/html/modules/AOS_PDF_Templates/PDF_Lib/classes/cssmgr.php on line 1326
[Thu Sep 09 15:35:03.670319 2021] [php7:warn] [pid 996] [client 10.100.106.71:44960] PHP Warning:  Illegal string offset 'ID' in /var/www/html/modules/AOS_PDF_Templates/PDF_Lib/classes/cssmgr.php on line 1610
[Thu Sep 09 15:35:03.670323 2021] [php7:warn] [pid 996] [client 10.100.106.71:44960] PHP Warning:  Illegal string offset 'ID' in /var/www/html/modules/AOS_PDF_Templates/PDF_Lib/classes/cssmgr.php on line 1614
```
